lib/rmg/CRmgTemplateZone.cpp

```diff
@@ -1463,6 +1495,37 @@ bool CRmgTemplateZone::createRequiredObjects()
 		}
 	}
 
+	//create nearby objects (e.g. extra resources close to mines)
+	for(const auto &object : nearbyObjects)
+	{
+		auto obj = object.first;
+		std::set<int3> possiblePositions;
+		for (auto blockedTile : object.second->getBlockedPos())
+		{
+			gen->foreachDirectNeighbour(blockedTile, [this, &possiblePositions](int3 pos)
+			{
+				if (!gen->isBlocked(pos))
+				{
+					//some resources still could be unaccessible, at least one free cell shall be
+					gen->foreach_neighbour(pos, [this, &possiblePositions, &pos](int3 p)
+												{
+						if(gen->isFree(p))
+							possiblePositions.insert(pos);
+					});
+				}
+			});
+		}
+
+		if(possiblePositions.empty())
+		{
+			delete obj; //is it correct way to prevent leak?
+			continue;
+		}
```

Comment:
Use `if ... else` to make it clearer when we're handling the object or deleting it.

Reply:
fixed